### ISSUE TITLE
n latest curl: (23) Failed writing received data to disk/application

### ISSUE THREAD

**Issue by p-d-boiko** [tj/n#798]
# Bug Report

command `n latest` fails with the error message:
```
curl: (23) Failed writing received data to disk/application

Error: failed to download version index (https://nodejs.org/dist/index.tab)
```

## Steps to Reproduce

In linux CLI run command `n latest` or `n lts`

## Expected Behaviour

Requested version of node should be downloaded and installed

## Actual Behaviour

The command fails writing downloaded version index file

## Other Information

`n` is installed in user's home directory, so user permissions is unlikely reason

## Configuration Details

```bash
$ n --version
v9.2.1

$ command -v node
/home/user/.local/software/n/bin/node

$ node -p process.platform
linux

$ which n
/home/user/.npm-global/bin/n
```

I would appreciate some hints how to find out where curl is attempting to store the version index file

**Comment by shadowspawn** [comment]
The index is not saved to disk, it is piped. The actual commands being run are:

```
curl --silent --compressed --location --fail --show-error --progress-bar https://nodejs.org/dist/index.tab | tail -n +2 | cut -f 1,3,10
```

```bash
$ curl --silent --compressed --location --fail --show-error --progress-bar https://nodejs.org/dist/index.tab | tail -n +2 | cut -f 1,3,10
v21.7.1	aix-ppc64,headers,linux-arm64,linux-armv7l,linux-ppc64le,linux-s390x,linux-x64,osx-arm64-tar,osx-x64-pkg,osx-x64-tar,src,win-arm64-7z,win-arm64-zip,win-x64-7z,win-x64-exe,win-x64-msi,win-x64-zip,win-x86-7z,win-x86-exe,win-x86-msi,win-x86-zip	-
v21.7.0	aix-ppc64,headers,linux-arm64,linux-armv7l,linux-ppc64le,linux-s390x,linux-x64,osx-arm64-tar,osx-x64-pkg,osx-x64-tar,src,win-arm64-7z,win-arm64-zip,win-x64-7z,win-x64-exe,win-x64-msi,win-x64-zip,win-x86-7z,win-x86-exe,win-x86-msi,win-x86-zip	-
v21.6.2	aix-ppc64,headers,linux-arm64,linux-armv7l,linux-ppc64le,linux-s390x,linux-x64,osx-arm64-tar,osx-x64-pkg,osx-x64-tar,src,win-arm64-7z,win-arm64-zip,win-x64-7z,win-x64-exe,win-x64-msi,win-x64-zip,win-x86-7z,win-x86-exe,win-x86-msi,win-x86-zip	-
...
```

**Comment by p-d-boiko** [comment]
curl in my Debian/Unstable fails with `--compressed` option, and works fine without it. When I remove this option on line 856 of the `n` script, it runs successfully. So the culprit is curl. There is an [issue and PR on their github](https://github.com/curl/curl/issues/13209). Waiting for it in my Debian package

```
$ curl --version
curl 8.7.1 (x86_64-pc-linux-gnu) libcurl/8.7.1 OpenSSL/3.1.5 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.7 libpsl/0.21.2 libssh2/1.11.0 nghttp2/1.59.0 librtmp/2.3 OpenLDAP/2.5.16
Release-Date: 2024-03-27, security patched: 8.7.1-1
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```

**Comment by shadowspawn** [comment]
I saw curl 8.7.1 is available in Homebrew, so getting into common distributions. Released n 9.2.3 which removes use of the `--uncompressed` option for now (#801).
